components/scream/src/share/atm_process/atmosphere_process_group.cpp

```diff
@@ -190,6 +190,106 @@ gather_internal_fields  () {
+  for (auto atm_proc : m_atm_processes) {
+
+    // We only want the checks on an individual processes. If
+    // atm_proc is a group, recursively call this function.
+    auto atm_proc_group = std::dynamic_pointer_cast<AtmosphereProcessGroup>(atm_proc);
+    if (atm_proc_group) {
+      // Require that a group does not enable conservation checks.
+      // Individual processes are the only ones that can define
+      // this check as there is currenty no concept of boundary
+      // fluxes over multiple processes implemented in the model.
+      EKAT_REQUIRE_MSG(not atm_proc_group->has_column_conservation_check(),
+                       "Error! The ATM process group \"" + atm_proc_group->name() + "\" attempted to enable "
+                       "conservation checks. Should have enable_column_conservation_checks=false for all "
+                       "process groups.\n");
+
+      atm_proc_group->setup_column_conservation_checks(conservation_check);
+      continue;
+    }
+
+    // For individual processes, first query if the checks are enabled.
+    // If not, continue to the next process.
+    if (not atm_proc->has_column_conservation_check()) {
+      continue;
+    }
+
+    // Since the checker is column local, require that an atm
+    // process that enables the check is a Physics process.
+    EKAT_REQUIRE_MSG(atm_proc->type() == AtmosphereProcessType::Physics,
+                     "Error! enable_column_conservation_checks=true "
+                     "for non-physics process \"" + atm_proc->name() + "\". "
+                     "This check is column local and therefore can only be run "
+                     "on physics processes.\n");
+ 
+    // Query the computed fields for this atm process and see if either the mass or energy computation
+    // might be changed after the process has run. If no field used in the mass or energy calculate
+    // is updated by this process, there is no need to run the check.
+    const std::string phys_grid_name  = conservation_check->get_grid()->name();
+    const bool updates_static_energy  = atm_proc->has_computed_field("T_mid", phys_grid_name);
+    const bool updates_kinetic_energy = atm_proc->has_computed_field("horiz_winds", phys_grid_name);
+    const bool updates_water_vapor    = atm_proc->has_computed_field("qv", phys_grid_name);
+    const bool updates_water_liquid   = atm_proc->has_computed_field("qc", phys_grid_name) ||
+                                        atm_proc->has_computed_field("qr", phys_grid_name);
+    const bool updates_water_ice      = atm_proc->has_computed_field("qi", phys_grid_name);
+    const bool mass_or_energy_is_updated = updates_static_energy || updates_kinetic_energy ||
+                                           updates_water_vapor   || updates_water_liquid ||
+                                           updates_water_ice;
+    EKAT_REQUIRE_MSG(mass_or_energy_is_updated, "Error! enable_column_conservation_checks=true for "
```

Comment:
@ambrad Should I instead throw a warning in the atm process group log here instead of throwing an error? Technically there is nothing wrong with running the check on a process that doesn't update mass or energy, it just wastes time. Radiation for instance doesn't update mass, but still runs the check on mass since it needs the check on energy.

Maybe this is overkill.